### PR TITLE
set downtime on nagios

### PR DESCRIPTION
**Pull request by Hoshiyo**

### CODE REVIEW

File: fabfile/utils.py [hunk 1]
```diff
@@ -424,3 +424,52 @@ def show_time_deploy(td, show=False):
         print(yellow(time_deploy))
 
     return time_deploy
+
+
+@roles('tyr_master')
+def downtime_request(host, start_date, end_date, service):
```
Comment (kinnou02) [inline fabfile/utils.py]: Maybe we can abstract thing a little bit. So we will be able to use something else latter.

File: fabfile/utils.py [hunk 2]
```diff
@@ -424,3 +424,52 @@ def show_time_deploy(td, show=False):
...
+    """ request for the downtime """
+    message = 'Deployment'
+
+    run('curl --silent --show-error '
```
Comment (kinnou02) [inline fabfile/utils.py]: There is no python client for this api?

**Comment by antoine-de** [comment]
@Hoshiyo you didn't add the context manager ?

File: fabfile/utils.py [hunk 3]
```diff
@@ -424,3 +429,80 @@ def show_time_deploy(td, show=False):
...
     return time_deploy
+
+
+class SupervisionHandler(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def stop_supervision(self, host, service, duration=None):
+        """
+        TODO
```
Comment (antoine-de) [inline fabfile/utils.py]: you need to add the comment :wink:

File: fabfile/utils.py [hunk 4]
```diff
@@ -424,3 +429,80 @@ def show_time_deploy(td, show=False):
...
+    def format_host(self, host):
+        """the nagio supervisor only wants a short host name"""
+        move_host = re.compile(r'@(.*?)\.')
+        return move_host.search(host).group(1)
+
+    def stop_supervision(self, host, service, duration=None):
+        start_dt = time.time()
+        start_str = datetime.datetime.fromtimestamp(start_dt).strftime('%d-%m-%Y %H:%M:%S')
+        if duration is None:
+            print("ERROR: 'duration' attribute cannot be a NoneType")
```
Comment (antoine-de) [inline fabfile/utils.py]: you need to raise an error there, we want to stop, not just skip the supervision monitoring

File: fabfile/utils.py [hunk 5]
```diff
@@ -424,3 +429,80 @@ def show_time_deploy(td, show=False):
...
+        self.user = user
+        self.pwd = pwd
+
+    def format_host(self, host):
+        """the nagio supervisor only wants a short host name"""
+        move_host = re.compile(r'@(.*?)\.')
+        return move_host.search(host).group(1)
+
+    def stop_supervision(self, host, service, duration=None):
+        start_dt = time.time()
```
Comment (antoine-de) [inline fabfile/utils.py]: you should transform it to a datetime here, and add a timedelta afterward

File: fabfile/env/platforms.py [hunk 6]
```diff
@@ -289,6 +289,26 @@
 
 env.is_prod = False
 
+# use for supervision
+env.supervision_handler = None
+env.supervision_config = dict(
+    tyr_beat=dict(
+        downtime=0,
```
Comment (antoine-de) [inline fabfile/env/platforms.py]: I think it's clearer if the downtime is a timedelta.

why do you need this dict ?

**Pull request merged**